ApiSurface/Test/TestVersionFile.fs

```diff
@@ -120,3 +122,48 @@ module TestVersionFile =
     [<TestCaseSource "versionFiles">]
     let ``Can write version file`` (versionFile : VersionFile) =
         versionFile |> serialise |> parse |> shouldEqual versionFile
+
+    let versionFileGen : Gen<VersionFile> =
+        gen {
+            let! major = Arb.generate<int>
+            let major = abs major
+            let! minor = Arb.generate<int>
+            let minor = abs minor
+            let! releaseRefSpec = Arb.generate<NonNull<string> list>
+            let! pathFilters = Arb.generate<NonNull<string> list option>
+
+            return
+                {
+                    Version = sprintf "%i.%i" major minor
+                    PublicReleaseRefSpec = releaseRefSpec |> List.map (fun spec -> spec.Get)
+                    PathFilters = pathFilters |> Option.map (List.map (fun filter -> filter.Get))
+                }
+        }
+
+    let private incrementVersion (version : string) =
+        match version.Split '.' with
+        | [| major ; _minor |] -> sprintf "%i.0" (int major + 1)
+        | _ -> failwithf "Unrecognised version string: %s" version
+
+    [<Test>]
+    let ``version JSON file can be written`` () =
```

Comment:
The name similarity to the test confused me for a second.

More importantly, I couldn't quite figure out what was wrong with the test in it's present form (and I don't have .NET 3.1 installed, but I locally changed the test project to only target .NET 7). It is just that the generated version data vs the update `ApiSurface.updateVersionJson` could potentially lead to lines from the previous write remaining, creating invalid JSON?

Reply:
Yes. The file is not truncated before writing, on `main`, so old parts of the end of the file may remain after we've overwritten it.